### PR TITLE
Rudimentary documentation of ocamlnat (PR#7254)

### PR DESCRIPTION
**Pull request by mshinwell**
[PR#7254](https://caml.inria.fr/mantis/view.php?id=7254) ended up in a request for some basic documentation of ocamlnat, which this pull request provides.

### CODE REVIEW

File: manual/manual/cmds/top.etex [hunk 1]
```diff
@@ -394,6 +394,25 @@ not opened, though; you still have to do
...
+
+{\bf This section describes a tool that is not yet officially supported %
+but may be found useful.}
+
+OCaml code executing in the traditional toplevel system uses the bytecode
+interpreter.  When increased performance is required, or for testing
+programs that will only execute correctly when compiled to native code,
+the {\em native toplevel} may be used instead.
+
+From the built source tree of the OCaml distribution you may use
```
Comment (Octachron) [inline manual/manual/cmds/top.etex]: I think that it is important to tell the reader explicitely that `ocamlnat` is not part of standard OCaml installations; and that installing it requires the compiler source files —which is of course implicit in the following, but the repetition would make more apparent the technical requirements of this section.

**Comment by gasche** [comment]
Maybe having ` (Experimental)` in the section title could help clarify the status of this part of the documentation.

**Comment by mshinwell** [comment]
Comments taken into account

**Comment by shindere** [comment]
It requires the source files, sure, but it also requires some special
tricks to be compiled. Perhaps it could be said that it is the `make
ocamlnat` command that can be used?

In the long run, it may be better to have a configure option to just
enable native toplevel. If this option is given, ocamlnat could then be
compiled and installed as part of the regular compile / install process.

**Comment by mshinwell** [comment]
@shindere I did mention "make ocamlnat".  Agreed that seems a sensible longer-term plan.

**Comment by shindere** [comment]
OK the PR looks good to me, thanks!

File: manual/manual/cmds/top.etex [hunk 2]
```diff
@@ -394,6 +394,28 @@ not opened, though; you still have to do
...
+From the built source tree of the distribution you may use
+{\tt make natruntop} to build and execute a native toplevel.  (Alternatively
+{\tt make ocamlnat} can be used, which just performs the build step.)
+
+If the {\tt make install} command is run after having built the native
+toplevel then the {\tt ocamlnat} program (either from the source or the
+installation directory) may be invoked directly rather than using
+{\tt make natruntop}.
+
 \section{Options}
```
Comment (Octachron) [inline manual/manual/cmds/top.etex]: I missed this point before rereading the manual, but the `Options` section here are `ocamlmktop`' s options. It should be moved directly after the `ocamlmktop` section (maybe as a subsection?) . Sorry for missing this point earlier.

Comment (mshinwell) [reply manual/manual/cmds/top.etex]: Well spotted.  I've kept ocamlnat as a section but changed ocamlmktop's options to a subsection.

**Pull request merged**